plugin-testlib/src/main/kotlin/io/spine/tools/gradle/testing/RunnerArguments.kt

```diff
@@ -0,0 +1,126 @@
+ */
+package io.spine.tools.gradle.testing
+
+import io.spine.tools.gradle.task.TaskName
+
+/**
+ * Create Gradle Runner arguments for a task.
+ */
+internal class RunnerArguments internal constructor(
+
+    /** If `true`, [CliOption.debug] will be passed to the runner.  */
+    private val debug: Boolean = false,
+
+    /** If `true`, [CliOption.stacktrace] will be passed to the runner. */
+    private val stacktrace: Boolean = true,
+
+    /** If `true`, [CliOption.noDaemon] will be passed to the runner. */
+    private val noDaemon: Boolean = false,
+
+    /** Properties passed to the runner. */
+    private val properties: Map<String, String> = mapOf()
+) {
+
+    /** Turns on the debug flag. */
+    fun withDebugLogging(): RunnerArguments {
+        return RunnerArguments(
+            debug = true,
+            stacktrace = this.stacktrace,
+            noDaemon = this.noDaemon,
+            properties = this.properties
+        )
+    }
+
+    /** Turns off the stacktrace output. */
+    fun noStacktrace(): RunnerArguments {
+        return RunnerArguments(
+            debug = this.debug,
+            stacktrace = false,
+            noDaemon = this.noDaemon,
+            properties = this.properties
+        )
+    }
+
+    /** Turns on the `--no-daemon` flag. */
+    fun noDaemon(): RunnerArguments {
+        return RunnerArguments(
+            debug = this.debug,
+            stacktrace = this.stacktrace,
+            noDaemon = true,
+            properties = this.properties
+        )
+    }
+
+    /** Adds a Gradle property entry to the command line arguments. */
+    fun withProperty(name: String, value: String): RunnerArguments {
+        require(name.isNotEmpty())
+        require(name.isNotBlank())
+        require(value.isNotBlank())
+        return RunnerArguments(
+            debug = this.debug,
+            stacktrace = this.stacktrace,
+            noDaemon = this.noDaemon,
+            properties = this.properties + Pair(name, value)
+        )
+    }
+
+    /** Adds passed properties to the arguments. */
+    fun withProperties(properties: Map<String, String>): RunnerArguments{
+        return RunnerArguments(
+            debug = this.debug,
+            stacktrace = this.stacktrace,
+            noDaemon = this.noDaemon,
+            properties = this.properties + properties
+        )
+    }
+
+    /**
+     * Adds the passed properties to those that may be already applied and returns
+     * resulting array of command line arguments.
+     */
+    fun forTask(task: TaskName): List<String> {
+        val args: MutableList<String> = taskWithOptions(task)
+        properties.forEach { entry ->
+            args.add(CliProperty(entry).argument())
+        }
+        return args.toList()
+    }
+
+    private fun taskWithOptions(task: TaskName): MutableList<String> {
+        val args: MutableList<String> = mutableListOf(task.name())
+        if (debug) {
+            args.add(CliOption.debug.argument())
```

Comment:
Since we're making the `CliOption` extendable, it'd be great if users could add custom ones. Just like with properties.